client/my-sites/site-settings/publishing-tools/index.jsx

```diff
@@ -0,0 +1,172 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import SectionHeader from 'components/section-header';
+import Card from 'components/card';
+import Button from 'components/button';
+import JetpackModuleToggle from '../jetpack-module-toggle';
+import FormLegend from 'components/forms/form-legend';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isJetpackModuleActive } from 'state/selectors';
+import { regeneratePostByEmail } from 'state/jetpack/settings/actions';
+import { isRegeneratingPostByEmail } from 'state/jetpack/settings/selectors';
+import InfoPopover from 'components/info-popover';
+import ExternalLink from 'components/external-link';
+import ClipboardButtonInput from 'components/clipboard-button-input';
+import PressThis from '../press-this';
+
+class PublishingTools extends Component {
+	componentDidUpdate() {
+		const {
+			fields,
+			postByEmailAddressModuleActive,
+			regeneratingPostByEmail,
+			selectedSiteId
+		} = this.props;
+
+		if ( postByEmailAddressModuleActive && regeneratingPostByEmail === null && ! fields.post_by_email_address ) {
+			this.props.regeneratePostByEmail( selectedSiteId );
+		}
+	}
+
+	onRegenerateButtonClick = () => {
+		this.props.regeneratePostByEmail( this.props.selectedSiteId );
+	}
+
+	isFormPending() {
+		const {
+			isRequestingSettings,
+			isSavingSettings,
+		} = this.props;
+
+		return isRequestingSettings || isSavingSettings;
+	}
+
+	renderPostByEmailSettings() {
+		const { fields, translate, postByEmailAddressModuleActive, regeneratingPostByEmail } = this.props;
+		const isFormPending = this.isFormPending();
+		const email = fields.post_by_email_address && fields.post_by_email_address !== 'regenerate' ? fields.post_by_email_address : '';
+		const labelClassName = regeneratingPostByEmail || ! postByEmailAddressModuleActive ? 'is-disabled' : null;
+
+		return (
+			<div className="publishing-tools__module-settings is-indented">
+				<FormLabel className={ labelClassName }>
+					{ translate( 'Email Address' ) }
+				</FormLabel>
+				<ClipboardButtonInput
+					className="publishing-tools__email-address"
+					disabled={ regeneratingPostByEmail || ! postByEmailAddressModuleActive }
+					value={ email }
+				/>
+				<Button
+					compact
+					onClick={ this.onRegenerateButtonClick }
+					disabled={ isFormPending || regeneratingPostByEmail || ! postByEmailAddressModuleActive }
+				>
+					{ regeneratingPostByEmail
+						? translate( 'Regenerating…' )
+						: translate( 'Regenerate address' )
```

Comment:
Thanks for dropping by, @yoavf! Well, we can, but it will probably not be specific enough. @MichaelArestad do you think we can replace it with the suggested one? I'm fine with replacing it, but didn't want to deviate from the design mockups.